TypeCobol/Compiler/CodeElements/Expressions/Expression.cs

```diff
@@ -201,20 +257,90 @@ public override bool AcceptASTVisitor(IASTVisitor astVisitor)
-		GreaterThan,
-		GreaterThanOrEqualTo,
-		LessThan,
-		LessThanOrEqualTo,
-		NotEqualTo
-	}
+    public class RelationalOperator : IVisitable
+    {
+        public static RelationalOperator EqualTo = new RelationalOperator(new SyntaxProperty<RelationalOperatorSymbol>(RelationalOperatorSymbol.EqualTo, null));
+        public static RelationalOperator GreaterThan = new RelationalOperator(new SyntaxProperty<RelationalOperatorSymbol>(RelationalOperatorSymbol.GreaterThan, null));
+        public static RelationalOperator GreaterThanOrEqualTo = new RelationalOperator(new SyntaxProperty<RelationalOperatorSymbol>(RelationalOperatorSymbol.GreaterThanOrEqualTo, null));
+        public static RelationalOperator LessThan = new RelationalOperator(new SyntaxProperty<RelationalOperatorSymbol>(RelationalOperatorSymbol.LessThan, null));
+        public static RelationalOperator LessThanOrEqualTo = new RelationalOperator(new SyntaxProperty<RelationalOperatorSymbol>(RelationalOperatorSymbol.LessThanOrEqualTo, null));
+        public static RelationalOperator NotEqualTo = new RelationalOperator(new SyntaxProperty<RelationalOperatorSymbol>(RelationalOperatorSymbol.NotEqualTo, null));
 
-	/// <summary>
+        public Token NotToken { get; private set; }
+        public SyntaxProperty<RelationalOperatorSymbol> RelationalOperatorSyntaxProperty { get; private set; }
+
+        public RelationalOperator(SyntaxProperty<RelationalOperatorSymbol> relationalOperatorSymbol)
+        {
+            RelationalOperatorSyntaxProperty = relationalOperatorSymbol;
+        }
+
+        public RelationalOperator(SyntaxProperty<RelationalOperatorSymbol> relationalOperatorSymbol, Token notToken) : 
+            this(relationalOperatorSymbol)
+        {
+            NotToken = notToken;
+        }
+
+        public bool SemanticEqualTo(RelationalOperator other)
+        {
+            // Both with NOT or both without NOT
+            if ((NotToken != null && other.NotToken != null) || (NotToken == null && other.NotToken == null))
+            {
+                return RelationalOperatorSyntaxProperty.Value == other.RelationalOperatorSyntaxProperty.Value;
+            }
+
+            return RelationalOperatorSyntaxProperty.Value.Opposite() == other.RelationalOperatorSyntaxProperty.Value;
+        }
+
+        public bool AcceptASTVisitor(IASTVisitor astVisitor)
+        {
+            return astVisitor.Visit(this)
+                   && this.ContinueVisitToChildren(astVisitor, NotToken, RelationalOperatorSyntaxProperty);
+        }
+
+        public override string ToString()
+        {
+            return (NotToken == null ? "" : NotToken.Text + " ") + RelationalOperatorSyntaxProperty.Value;
+        }
+    }
+
+    /// <summary>
+    /// Relational operators and their meanings
+    /// </summary>
+    public enum RelationalOperatorSymbol
+    {
+        EqualTo,
+        GreaterThan,
+        GreaterThanOrEqualTo,
+        LessThan,
+        LessThanOrEqualTo,
+        NotEqualTo
```

Comment:
`NotEqualTo` should be removed now.

Reply:
This is required to find the opposite of a RelationalOperatorSymbol, and for the SemanticEquals().

Reply:
Ok so after discussing this, we keep this `NotEqualTo` with a comment explaining that it is not tied to a syntax in source but use it for a new `SemanticOperator` property which returns the "normalized" operator integrating the NOT token if any.